# HG changeset patch
# User tailor@segfault.lan
# Date 1198875851 0
#      Fri Dec 28 21:04:11 2007 +0000
# Branch release-3-0-x
# Node ID a18fac5c789dd3c7c13eaa770e530b869a8ecb72
# Parent  58f5fab3ebe5be60a39addd9b5bee8ac9e742836
Tailorization
Import of the upstream sources from
 Repository: /scratch/jwe/mercurial/cvs
       Kind: cvs
     Module: octave
   Revision: 2007-12-28 21:04:11 by jwe

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2007-12-28  Kai Habel  <kai.habel@gmx.de>
+
+	* plot/pcolor.m: Swap 1st and 2nd argument in call to meshgrid.
+	Remove unnecessary call of size function.
+
 2007-12-21  John W. Eaton  <jwe@octave.org>
 
 	Version 3.0.0 released.
 
 2007-12-21  Soren Hauberg  <hauberg@gmail.com>
 
 	* image/imshow.m: Accept empty value for display_range.
 
diff --git a/scripts/plot/pcolor.m b/scripts/plot/pcolor.m
--- a/scripts/plot/pcolor.m
+++ b/scripts/plot/pcolor.m
@@ -31,19 +31,19 @@
 ## Author: Kai Habel <kai.habel@gmx.de>
 
 function h = pcolor (x, y, c)
 
   newplot ();
 
   if (nargin == 1)
     c = x;
-    z = zeros (size (c));
-    [nr, nc] = size (c);
-    [x, y] = meshgrid (1:nr, 1:nc);
+    [nr, nc] = size(c);
+    z = zeros (nr, nc);
+    [x, y] = meshgrid (1:nc, 1:nr);
   elseif (nargin == 3)
     z = zeros (size (c));
   else
     print_usage ();
   endif
 
   tmp = surface (x, y, z, c);
 
